pkg/watermark/generic/jetstream/generic.go

```diff
@@ -50,6 +50,7 @@ func BuildWatermarkProgressors(ctx context.Context, vertexInstance *v1alpha1.Ver
 
 	var fetchWatermark fetch.Fetcher
 	if fromBuffer.Type == v1alpha1.SourceBuffer {
+		// TODO: does it use a non-nil hbWatch?
```

Comment:
Meaning...
is this really executed for source vertex?
https://github.com/numaproj/numaflow/blob/e953699e321baae2c46af9e7b2bce56b5ba935ce/pkg/watermark/fetch/processor_manager.go#L55-L57

Reply:
also #230

Reply:
can you check the source's code and see whether we need this check?

Reply:
👀